### PR TITLE
Requestfix counterpart

### PR DESCRIPTION
**Pull request by CoMPaTech**
Note: already changed translation handling per 109 next week

### CODE REVIEW

File: custom_components/plugwise-beta/__init__.py [hunk 1]
```diff
@@ -36,9 +37,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     )
 
     try:
-        await api.connect()
+        connected = await api.connect()
+
+        if not connected:
+            _LOGGER.error("Unable to Smile: %s",api.smile_status)
```
Comment (bouwew) [inline custom_components/plugwise-beta/__init__.py:43]: Unable to connect to Smile? Seems a word is missing.

File: custom_components/plugwise-beta/__init__.py [hunk 2]
```diff
@@ -148,14 +154,17 @@ def async_remove_listener(self, update_callback):
...
+            _LOGGER.error("PSmile has no listeners, not updating")
             return
 
-        _LOGGER.debug("Plugwise Smile updating data using: %s", self.update_method)
+        _LOGGER.debug("Smile updating data using: %s", self.update_method)
 
-        await self.api.full_update_device()
+        update_success, update_status = await self.api.full_update_device()
+        if not update_success:
+            _LOGGER.error("Smile update failed: %s", update_status)
```
Comment (bouwew) [inline custom_components/plugwise-beta/__init__.py:166]: Raise a PlatformNotReady here as well?

File: custom_components/plugwise-beta/__init__.py (hunk 1 above)
Comment (CoMPaTech) [reply custom_components/plugwise-beta/__init__.py]: fixed

File: custom_components/plugwise-beta/__init__.py (hunk 2 above)
Comment (CoMPaTech) [reply custom_components/plugwise-beta/__init__.py:166]: From what I found in HA-core homeassistant/helpers/entity_platform.py it's only used during setup, not on updateloops?

Comment (bouwew) [reply custom_components/plugwise-beta/__init__.py:166]: Ok, understood.

**Comment by CoMPaTech** [comment]
Counterpart of https://github.com/plugwise/Plugwise-Smile/pull/41

Now rewritten with exceptions instead of status' and returns

**Pull request merged**